In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive




---



---


 # ***EDA:***


---



---






### Importar librerias

In [1]:
# Importar librerias necesarias
import pandas as pd
import pyarrow.parquet as pq
import pyarrow as py






---



---

## **1_ User Reviews**



---

### Carga del DataFrame 'df_user_reviews' comprimido




In [2]:
#Se carga el archivo comprimido
df_user_reviews = pd.read_parquet('/content/drive/MyDrive/Colab Notebooks/PI_1/Data/df_user_reviews_sentimientos.parquet')




---


### **Descripción General:**

In [3]:
df_user_reviews.shape
print(f'El DataFrame posee la cnatidad de Filas igual a:{df_user_reviews.shape[0]} y de Columnas : {df_user_reviews.shape[1]}')

El DataFrame posee la cnatidad de Filas igual a:59305 y de Columnas : 10


In [4]:
df_user_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59305 entries, 0 to 59304
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Id_Usuario             59305 non-null  object
 1   Url_Usuario            59305 non-null  object
 2   funny                  59305 non-null  object
 3   Posteo                 59305 non-null  object
 4   Ultima_Edicion         59305 non-null  object
 5   Id_Item                59305 non-null  object
 6   Util                   59305 non-null  object
 7   Recomendado            59305 non-null  object
 8   clean_reviews          59305 non-null  object
 9   analisis_sentimientos  59305 non-null  int64 
dtypes: int64(1), object(9)
memory usage: 4.5+ MB


  El conjunto de datos df_user_reviews contiene información sobre reseñas de usuarios en la plataforma Steam. Con 59305 entradas y 10 columnas, proporciona detalles sobre el usuario (Id_Usuario y Url_Usuario), elementos revisados (Id_Item), fechas de publicación y edición, así como la opinión del usuario , la utilidad y recomendación de la reseña.


In [5]:
# Visualizacion general del DataFrame
df_user_reviews.head()

Id_Usuario                                        Url_Usuario funny  \
0  76561197970982479  http://steamcommunity.com/profiles/76561197970...         
1  76561197970982479  http://steamcommunity.com/profiles/76561197970...         
2  76561197970982479  http://steamcommunity.com/profiles/76561197970...         
3            js41637               http://steamcommunity.com/id/js41637         
4            js41637               http://steamcommunity.com/id/js41637         

                    Posteo Ultima_Edicion Id_Item  \
0   posted november 5 2011                   1250   
1      posted july 15 2011                  22200   
2     posted april 21 2011                  43110   
3      posted june 24 2014                 251610   
4  posted september 8 2013                 227300   

                                              Util Recomendado  \
0                                   no ratings yet        true   
1                                   no ratings yet        true   
2                                   no ratings yet        true   
3  15 of 20 people (75%) found this review helpful        true   
4     0 of 1 people (0%) found this review helpful        true   

                                       clean_reviews  analisis_sentimientos  
0  simple yet great replayability opinion zombie ...                      2  
1                           unique worth playthrough                      2  
2  great atmosphere gunplay bit chunky time end d...                      2  
3  know think see title barbie dreamhouse party i...                      2  
4  simple actually simple truck driving simulator...                      0







---


### **Descripcion del contenido de las columnas**

- 'Id_Usuario': Identificador unico de usuario.
- 'Url_Usuario': URL perfil del usuario.
- 'funny': Divirtio el comentario posteado.
- 'Posteo': Fecha de posteo.
- 'Ultima_Edicion': Ultima modificacion del posteo.
- 'Id_Item': Identificador del juego.
- 'Util': Porcentajes de personas que reconocen util el posteo.       
- 'Recomendado': Recomendacion del juego como Verdadero o Falso.
- 'clean_reviews': Comonetario sobre el juego.
- 'analisis_sentimientos': toma el valor '0' si es malo, '1' si es neutral y '2' si es positivo sobre el analisis de la columna 'clean_reviews'.








---


### **Variables**

In [6]:
df_user_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59305 entries, 0 to 59304
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Id_Usuario             59305 non-null  object
 1   Url_Usuario            59305 non-null  object
 2   funny                  59305 non-null  object
 3   Posteo                 59305 non-null  object
 4   Ultima_Edicion         59305 non-null  object
 5   Id_Item                59305 non-null  object
 6   Util                   59305 non-null  object
 7   Recomendado            59305 non-null  object
 8   clean_reviews          59305 non-null  object
 9   analisis_sentimientos  59305 non-null  int64 
dtypes: int64(1), object(9)
memory usage: 4.5+ MB


In [58]:
# Descripción de los resultados del Análisis de sentimientos
df_user_reviews['Analisis_sentimientos'].describe()


count    59305.000000
mean         1.284984
std          0.826885
min          0.000000
25%          1.000000
50%          2.000000
75%          2.000000
max          2.000000
Name: Analisis_sentimientos, dtype: float64

- La media es aproximadamente 1.28. Esto indica que, en promedio, las opiniones son más positivas que negativas, ya que el valor está por encima del punto medio.
- El primer cuartil (25%) es 1.0, el segundo cuartil (50%) es 2.0, y el tercer cuartil (75%) es 2.0. Esto sugiere que al menos el 50% de las opiniones tienen una puntuación de sentimiento de 2.0 (positivo) o superior.

In [47]:
# Cambiar el Valr de la Variable 'Id_Item'
df_user_reviews['Id_Item'] = df_user_reviews['Id_Item'].astype(int)








---


### Valores Nulos

In [8]:
valores_nulos = df_user_reviews.isna().sum()
print(f'La cantidad de valores Nulos que posee el DataFrame es: \n{valores_nulos}')

La cantidad de valores Nulos que posee el DataFrame es: 
Id_Usuario               0
Url_Usuario              0
funny                    0
Posteo                   0
Ultima_Edicion           0
Id_Item                  0
Util                     0
Recomendado              0
clean_reviews            0
analisis_sentimientos    0
dtype: int64


- Ninguna de las columnas poseen valores nulos







---


### Columnas

In [9]:
# Cambiar el nombre de las columnas
df_user_reviews = df_user_reviews.rename(columns={ 'clean_reviews': 'Comentarios', 'analisis_sentimientos': 'Analisis_sentimientos'})

In [11]:
# Extrae el año de la columna Fecha_Lanzamiento y crea columna Año_Lanzamiento
df_user_reviews['Año_Posteo'] = df_user_reviews['Posteo'].str.extract(r'(\d{4})', expand=False)

In [12]:
df_user_reviews.columns

Index(['Id_Usuario', 'Url_Usuario', 'funny', 'Posteo', 'Ultima_Edicion',
       'Id_Item', 'Util', 'Recomendado', 'Comentarios',
       'Analisis_sentimientos', 'Año_Posteo'],
      dtype='object')

In [36]:
# Eliminar columnas
df_user_reviews = df_user_reviews.drop(['Url_Usuario', 'funny', 'Posteo', 'Ultima_Edicion', 'Util',], axis=1)

- Se decide eliminar las columas 'Posteo', 'Ultima_Edicion', debido a que se crea una nueva, con los años unicamente, que serian necesario para el futuro analisis y estas columnas serian redundantes.
- Se decide eliminar las columas 'Util', 'Url_Usuario', 'funny', ya ue no aportan informacion necesaria para el analisis.

In [30]:
# Cauntificar los valores Nulos en la columna creada
df_user_reviews.isna().sum()

Id_Usuario                   0
Url_Usuario                  0
funny                        0
Posteo                       0
Ultima_Edicion               0
Id_Item                      0
Util                         0
Recomendado                  0
Comentarios                  0
Analisis_sentimientos        0
Año_Posteo               10119
dtype: int64

In [27]:
# Contar los Años de la columna 'Año_Posteo'
df_user_reviews['Año_Posteo'].value_counts()

2014    22111
2015    18470
2013     6792
2012     1215
2011      532
2010       66
Name: Año_Posteo, dtype: int64

In [31]:
# Rellenar los nulos con año 2016
df_user_reviews = df_user_reviews.fillna(2016)

In [33]:
# Convertir la columna 'Año_Posteo' en INT
df_user_reviews['Año_Posteo'] = df_user_reviews['Año_Posteo'].astype(int)

- Se imputan los nulos, como año 2016, toma solo los valores de lo meses y dias en el año que se tiene los datos.

In [48]:
df_user_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59305 entries, 0 to 59304
Data columns (total 6 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Id_Usuario             59305 non-null  object
 1   Id_Item                59305 non-null  int64 
 2   Recomendado            59305 non-null  object
 3   Comentarios            59305 non-null  object
 4   Analisis_sentimientos  59305 non-null  int64 
 5   Año_Posteo             59305 non-null  int64 
dtypes: int64(3), object(3)
memory usage: 2.7+ MB








---


### Valores duplicados

In [138]:
# Comprobar si hay valores duplicados y mostrar las filas duplicadas
print(f'Tamaño del set antes de eliminar las filas repetidas: {df_user_reviews.shape}')
df_user_reviews.drop_duplicates(inplace=True)
print(f'Tamaño del set después de eliminar las filas repetidas: {df_user_reviews.shape}')


Tamaño del set antes de eliminar las filas repetidas: (59305, 6)
Tamaño del set después de eliminar las filas repetidas: (58431, 6)








---


### Guardar el Nuevo DataFrame

In [139]:
# Guardar el DataFrame comprimido en Parquet
df_user_reviews.to_parquet('df_user_reviews_limpio.parquet')








---


---
## 2_ Steam Games



---

### Carga del DataFrame 'df_user_reviews' comprimido

In [2]:
#Se carga el archivo para trabajarlo
df_steam_games = pd.read_parquet('/content/drive/MyDrive/Colab Notebooks/PI_1/Data/df_steam_games.parquet')

---

### Descripcion General

In [3]:
df_steam_games.shape
print(f'El DataFrame posee la cnatidad de Filas igual a:{df_steam_games.shape[0]} y de Columnas : {df_steam_games.shape[1]}')

El DataFrame posee la cnatidad de Filas igual a:30067 y de Columnas : 14


In [4]:
df_steam_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30067 entries, 0 to 30066
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Empresa_Publicadora  30067 non-null  object 
 1   Generos              30067 non-null  object 
 2   Nombre_Contenido     30067 non-null  object 
 3   Titulo               30067 non-null  object 
 4   Url                  30067 non-null  object 
 5   Fecha_Lanzamiento    30067 non-null  object 
 6   Etiquetas            30067 non-null  object 
 7   Reviews_Contenido    30067 non-null  object 
 8   Especificaciones     30067 non-null  object 
 9   Precio               30067 non-null  float64
 10  Acceso_Temprano      30067 non-null  float64
 11  Id_Item              30067 non-null  Int64  
 12  Desarrollador        30067 non-null  object 
 13  Año_Lanzamiento      29966 non-null  Int64  
dtypes: Int64(2), float64(2), object(10)
memory usage: 3.3+ MB


- El conjunto de datos df_steam_games contiene información sobre juegos disponibles en la plataforma Steam. Consta de 30067 entradas y 14 columnas, proporcionando detalles sobre el editor (Empresa_Publicadora), Generos, Nombre_Contenido, Titulo, Fecha_Lanzamiento, Etiquetas, URL de reseñas, especificaciones, precio, Acceso_Temprano, Id_Item, Desarrollador y Año_Lanzamiento. Algunos de los datos son nulos en columna como Año_Lanzamiento. Este conjunto de datos presenta una variedad de información que puede ser analizada para obtener perspectivas interesantes sobre los juegos en la plataforma Steam.


---


### **Descripcion del contenido de las columnas**







---


### Variables

In [5]:
df_steam_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30067 entries, 0 to 30066
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Empresa_Publicadora  30067 non-null  object 
 1   Generos              30067 non-null  object 
 2   Nombre_Contenido     30067 non-null  object 
 3   Titulo               30067 non-null  object 
 4   Url                  30067 non-null  object 
 5   Fecha_Lanzamiento    30067 non-null  object 
 6   Etiquetas            30067 non-null  object 
 7   Reviews_Contenido    30067 non-null  object 
 8   Especificaciones     30067 non-null  object 
 9   Precio               30067 non-null  float64
 10  Acceso_Temprano      30067 non-null  float64
 11  Id_Item              30067 non-null  Int64  
 12  Desarrollador        30067 non-null  object 
 13  Año_Lanzamiento      29966 non-null  Int64  
dtypes: Int64(2), float64(2), object(10)
memory usage: 3.3+ MB


In [6]:
# Analizar las variables numericas Precio
df_steam_games[df_steam_games['Precio'] != -1][['Precio']].describe()


Precio
count  28820.000000
mean       8.870421
std       16.352657
min        0.000000
25%        2.990000
50%        4.990000
75%        9.990000
max      995.000000

- El precio promedio de los juegos es de aproximadamente 8.87 dolares y la desviación estándar es alta (16.35), indicando una variabilidad significativa en los precios. El precio mínimo es 0, lo que sugiere que algunos juegos son gratuitos. El 75% de los juegos tienen un precio igual o inferior a 9.99 unidades monetarias, cercano al del promedio. El precio máximo es 995 dolares lo que puede indicar a que es un juego Premiun, para desarrolladores.

In [7]:
df_steam_games['Año_Lanzamiento'].describe()

count        29966.0
mean     2014.878863
std        19.156818
min           1970.0
25%           2014.0
50%           2016.0
75%           2017.0
max           5275.0
Name: Año_Lanzamiento, dtype: Float64

In [8]:
# Se busca el juego con un año erroneo
año_erroneo = df_steam_games[df_steam_games['Año_Lanzamiento'] == 5275]
año_erroneo

Empresa_Publicadora                      Generos     Nombre_Contenido  \
12561              一次元创作组  casual, indie, early access  puzzle sisters foer   

                    Titulo                                                Url  \
12561  puzzle sisters foer  http://store.steampowered.com/app/710190/Puzzl...   

      Fecha_Lanzamiento                    Etiquetas  \
12561       预热群52756441  early access, casual, indie   

                                       Reviews_Contenido  \
12561  http://steamcommunity.com/app/710190/reviews/?...   

                                        Especificaciones  Precio  \
12561  single player, steam achievements, steam tradi...    -1.0   

       Acceso_Temprano  Id_Item Desarrollador  Año_Lanzamiento  
12561              1.0   710190        一次元创作组             5275

In [9]:
# Se elimina la fila con el año erroneo
df_steam_games = df_steam_games.drop(12561)







---


### Valores Nulos

In [10]:
df_steam_games.isna().sum()

Empresa_Publicadora      0
Generos                  0
Nombre_Contenido         0
Titulo                   0
Url                      0
Fecha_Lanzamiento        0
Etiquetas                0
Reviews_Contenido        0
Especificaciones         0
Precio                   0
Acceso_Temprano          0
Id_Item                  0
Desarrollador            0
Año_Lanzamiento        101
dtype: int64

In [11]:
# Filtrar las filas donde 'Año_Lanzamiento' es nulo
filas_sin_año = df_steam_games[df_steam_games['Año_Lanzamiento'].isna()]
filas_sin_año


Empresa_Publicadora  \
9                             qucheza   
36                               zha7   
6598                    ghost machine   
6950               strange loop games   
6994   alterego games, digital forest   
...                               ...   
29931                          8floor   
29936             dagestan technology   
29944                  maulidan games   
29951           new blood interactive   
30021  greyson richey, nicholas lives   

                                          Generos  Nombre_Contenido  \
9         action, indie, simulation, early access   uncanny islands   
36     adventure, free to play, rpg, early access    idle adventure   
6598                                action, indie          warhawks   
6950                 adventure, indie, simulation               eco   
6994                     adventure, casual, indie             woven   
...                                           ...               ...   
29931                            casual, strategy    lost artifacts   
29936                    action, adventure, indie           memento   
29944                            adventure, indie     spy of deimos   
29951                               action, indie              dusk   
30021           action, casual, indie, simulation  lonely astronaut   

                 Titulo                                                Url  \
9       uncanny islands  http://store.steampowered.com/app/768570/Uncan...   
36       idle adventure  http://store.steampowered.com/app/714020/Idle_...   
6598           warhawks  http://store.steampowered.com/app/380130/Warha...   
6950                eco      http://store.steampowered.com/app/382310/Eco/   
6994              woven    http://store.steampowered.com/app/504690/Woven/   
...                 ...                                                ...   
29931    lost artifacts  http://store.steampowered.com/app/708150/Lost_...   
29936           memento  http://store.steampowered.com/app/454940/Memento/   
29944     spy of deimos  http://store.steampowered.com/app/689820/Spy_o...   
29951              dusk     http://store.steampowered.com/app/519860/DUSK/   
30021  lonely astronaut  http://store.steampowered.com/app/755830/Lonel...   

      Fecha_Lanzamiento                                          Etiquetas  \
9                Soon..  early access, action, indie, simulation, survival   
36            Beta测试已开启         early access, free to play, adventure, rpg   
6598        Coming Soon                          action, indie, vr, flight   
6950        Coming Soon  survival, adventure, multiplayer, simulation, ...   
6994                TBA  indie, casual, adventure, singleplayer, story ...   
...                 ...                                                ...   
29931       Coming soon                                   strategy, casual   
29936               TBD                   indie, action, adventure, horror   
29944       Coming Soon                                   indie, adventure   
29951             SOON™  action, fps, indie, retro, shooter, horror, go...   
30021       Coming Soon                  casual, action, indie, simulation   

                                       Reviews_Contenido  \
9      http://steamcommunity.com/app/768570/reviews/?...   
36     http://steamcommunity.com/app/714020/reviews/?...   
6598   http://steamcommunity.com/app/380130/reviews/?...   
6950   http://steamcommunity.com/app/382310/reviews/?...   
6994   http://steamcommunity.com/app/504690/reviews/?...   
...                                                  ...   
29931  http://steamcommunity.com/app/708150/reviews/?...   
29936  http://steamcommunity.com/app/454940/reviews/?...   
29944  http://steamcommunity.com/app/689820/reviews/?...   
29951  http://steamcommunity.com/app/519860/reviews/?...   
30021  http://steamcommunity.com/app/755830/reviews/?...   

                                        Especificaciones  Precio  \
9        

In [12]:
# Calcular el porcentaje de filas con valores nulos en 'Año_Lanzamiento'
porcentaje_filas_sin_año = (len(filas_sin_año) / len(df_steam_games)) * 100

# Imprimir el resultado
print(f'El total de Filas sin año: {len(filas_sin_año)} \nEl total de Filas: {len(df_steam_games)} \nPorcentaje de filas sin año: {porcentaje_filas_sin_año:.2f}%')


El total de Filas sin año: 101 
El total de Filas: 30066 
Porcentaje de filas sin año: 0.34%


In [13]:
# Eliminar filas sin años del DataFrame
df_steam_games = df_steam_games.dropna(subset=['Año_Lanzamiento'])

- A ser el porcentaje bajo de datos sin año, con futuro lanzamiento, se proceden a eliminar, ya que son juegos que no salieron a la venta.







---


### Columnas

In [15]:
# Eliminar columnas
df_steam_games = df_steam_games.drop(['Empresa_Publicadora', 'Nombre_Contenido', 'Url', 'Fecha_Lanzamiento', 'Etiquetas', 'Reviews_Contenido', 'Acceso_Temprano',], axis=1)

- Se procede a eliminar las columnas Empresa_Publicadora', 'Url', 'Fecha_Lanzamiento', 'Etiquetas', 'Reviews_Contenido', 'Acceso_Temprano'debido a que tiene valores informacion redundante y no relevante para el analisis respectivo.







---


### Valores duplicados

In [17]:
# Se cuentan el total de duplicados y se los elimina
print(f'Tamaño del set antes de eliminar las filas repetidas: {df_steam_games.shape}')
df_steam_games.drop_duplicates(inplace=True)
print(f'Tamaño del set después de eliminar las filas repetidas: {df_steam_games.shape}')

Tamaño del set antes de eliminar las filas repetidas: (29965, 7)
Tamaño del set después de eliminar las filas repetidas: (29963, 7)








---


### Guardado del DataFrame

In [20]:
# Guardar el DataFrame comprimido en Parquet
df_steam_games.to_parquet('df_steam_games_limpio.parquet')






---



---


## **3_ User Items**

  
---

### Carga del DataFrame 'df_user_reviews' comprimido

In [33]:
# Se carga el archivo
df_user_items = pd.read_parquet('/content/drive/MyDrive/Colab Notebooks/PI_1/Data/df_user_items.parquet')

---

### Descripcion General

In [34]:
df_user_items.shape
print(f'El DataFrame posee la cantidad de Filas igual a:{df_user_items.shape[0]} y de Columnas : {df_user_items.shape[1]}')

El DataFrame posee la cantidad de Filas igual a:5153209 y de Columnas : 8


In [35]:
df_user_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5153209 entries, 0 to 5153208
Data columns (total 8 columns):
 #   Column                   Dtype  
---  ------                   -----  
 0   Id_Usuario               object 
 1   Conteo_Items             int64  
 2   Id_Steam                 object 
 3   Url_Usuario              object 
 4   Id_Item                  int64  
 5   Nombre_Juego             object 
 6   Tiempo_Jugado            float64
 7   Tiempo_Jugado_2_Semanas  float64
dtypes: float64(2), int64(2), object(4)
memory usage: 314.5+ MB


 El DataFrame df_user_items contiene información detallada sobre la actividad de los usuarios en relación con los ítems de un juego. Cada fila representa una entrada única que indica cuántas veces un usuario ha interactuado con un determinado ítem del juego y la cantidad de tiempo dedicado a ese ítem.




---


### **Descripcion del contenido de las columnas**


- Id_Usuario: Identificación única del usuario.
- Conteo_Items: Cantidad total de ítems que el usuario ha interactuado.
- Id_Steam: Identificación única del ítem.
- Url_Usuario: URL perfil del usuario.
- Id_Item: ID del juego.
- Nombre_Juego: Nombre del titulo
- Tiempo_Jugado: Tiempo total de juego dedicado al ítem por el usuario (en horas).
- Tiempo_Jugado_2_Semanas: Tiempo de juego dedicado al ítem en las últimas dos semanas (en horas).





---


### Variables

In [29]:
# Modificar el tipo de archivo de la columna 'Id_Item' a INT
df_user_items['Id_Item'] = df_user_items['Id_Item'].astype(int)

In [46]:
df_user_items[df_user_items['Tiempo_Jugado'] != 0]['Tiempo_Jugado'].describe()

count    3.285246e+06
mean     1.555251e+03
std      6.721032e+03
min      1.000000e+00
25%      4.400000e+01
50%      2.050000e+02
75%      8.080000e+02
max      6.427730e+05
Name: Tiempo_Jugado, dtype: float64

- La media de tiempo jugado es 1,555.25 horas, lo que indica que, en promedio, los usuarios han invertido bastante tiempo en los juegos.
- La desviación estándar de 6,721.03 horas sugiere una gran variabilidad en los datos. Algunos usuarios pueden jugar durante períodos extremadamente largos, mientras que otros juegan por períodos más cortos.
- El valor mínimo de 1 hora indica que hay usuarios que han jugado por un tiempo muy breve, posiblemente solo probando el juego.
- El 25° percentil de 44 horas indica que el 25% de los usuarios han jugado menos de 44 horas, mientras que el 75° percentil de 808 horas indica que el 75% de los usuarios han jugado menos de 808 horas.
- El máximo de 642,773 horas sugiere la presencia de valores atípicos significativos. Podrían ser usuarios extremadamente dedicados o posiblemente errores en la recopilación de datos. Se recomienda una revisión más detallada de estos casos.

---


### Valores Nulos

In [48]:
# Se buscan valores nulos
df_user_items.isna().sum()

Id_Usuario                 0
Conteo_Items               0
Id_Steam                   0
Url_Usuario                0
Id_Item                    0
Nombre_Juego               0
Tiempo_Jugado              0
Tiempo_Jugado_2_Semanas    0
dtype: int64

- No se observan Valores Nulos en las columnas.

---


### Columnas

In [50]:
# Eliminar columnas
df_user_items = df_user_items.drop(['Conteo_Items', 'Id_Steam', 'Url_Usuario'] , axis=1)

- Se elimina las columnas 'Conteo_Items', 'Id_Steam', 'Url_Usuario' debido a que tiene valores informacion redundante y no relevante para el analisis respectivo.

---


### Valores duplicados

In [54]:
# Se cuentan los valores duplicados
valores_duplicados = df_user_items.duplicated().sum()
print(f'Cantidad de valores duplicados: {valores_duplicados}')

Cantidad de valores duplicados: 59104


In [56]:
# Se eliminan los valores duplicados
print(f'Tamaño del set antes de eliminar las filas repetidas: {df_user_items.shape}')
df_user_items.drop_duplicates(inplace=True)
print(f'Tamaño del set después de eliminar las filas repetidas: {df_user_items.shape}')

Tamaño del set antes de eliminar las filas repetidas: (5153209, 5)
Tamaño del set después de eliminar las filas repetidas: (5094105, 5)


---


### Guardado del DataFrame

In [57]:
# Guardar el DataFrame comprimido en Parquet
df_user_items.to_parquet('df_user_items_limpio.parquet')